drivers/auxdisplay/auxdisplay_serlcd.c

```diff
@@ -0,0 +1,361 @@
+		data->cursor_y = row;
+
+		const uint8_t cursor_address = column + row_offsets[row];
+
+		return auxdisplay_serlcd_send_special_command(
+			dev, SERLCD_SPECIAL_SET_DD_RAM_ADDRESS | cursor_address);
+	}
+
+	/*
+	 * other types of movement are not implemented/supported
+	 */
+	return -ENOSYS;
+}
+
+static int auxdisplay_serlcd_cursor_position_get(const struct device *dev, int16_t *x, int16_t *y)
+{
+	const struct auxdisplay_serlcd_data *data = dev->data;
+
+	*x = (int16_t)data->cursor_x;
+	*y = (int16_t)data->cursor_y;
+
+	return 0;
+}
+
+static int auxdisplay_serlcd_capabilities_get(const struct device *dev,
+					      struct auxdisplay_capabilities *capabilities)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+
+	memcpy(capabilities, &config->capabilities, sizeof(struct auxdisplay_capabilities));
+
+	return 0;
+}
+
+static int auxdisplay_serlcd_clear(const struct device *dev)
+{
+	int rc = auxdisplay_serlcd_send_command(dev, SERLCD_COMMAND_CLEAR);
+
+	/*
+	 * clearing takes more time
+	 */
+	k_sleep(K_MSEC(10));
+	return rc;
+}
+
+static int auxdisplay_serlcd_custom_character_set(const struct device *dev,
+						  struct auxdisplay_character *character)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+
+	/*
+	 * only indexes 0..7 are supported
+	 */
+	uint8_t char_index = character->index & 0x07;
+
+	/*
+	 * custom characters are accessible via char codes 0x08..0x0f
+	 */
+	character->character_code = 0x08 & char_index;
+
+	int rc = auxdisplay_serlcd_send_command(dev, SERLCD_COMMAND_SET_CUSTOM_CHAR + char_index);
+
+	if (!rc) {
+		return rc;
+	}
+
+	/*
+	 * the display expects the custom character as 8 lines of 5 bit each, shades are not
+	 * supported
+	 */
+	for (int l = 0; l < 8; ++l) {
+		uint8_t buffer = 0;
+
+		for (int i = 0; i < 5; ++i) {
+			if (character->data[(l * 5) + i]) {
+				buffer |= BIT(4 - i);
+			}
+		}
+		rc = i2c_write_dt(&config->bus, &buffer, 1);
+		if (!rc) {
+			return rc;
+		}
+	}
+
+	return rc;
+}
+
+static int auxdisplay_serlcd_write(const struct device *dev, const uint8_t *text, uint16_t len)
+{
+	const struct auxdisplay_serlcd_config *config = dev->config;
+
+	int rc = 0;
+
+	/*
+	 * the display wraps around by itself, just write the text and update the position data
+	 */
+	for (int i = 0; i < len; ++i) {
+		uint8_t character = text[i];
+
+		/*
+		 * customer characters require a special command, so check for custom char
+		 */
+		if ((character & 0xf8) == 0x08) {
+			rc = auxdisplay_serlcd_send_command(dev, SERLCD_COMMAND_WRITE_CUSTOM_CHAR +
+									 (character & 0x07));
+			if (!rc) {
+				return rc;
+			}
+		} else if (character == SERLCD_BEGIN_COMMAND ||
+			   character == SERLCD_BEGIN_SPECIAL_COMMAND) {
+			/*
+			 * skip these characters in text, as they have a special meaning, if
+			 * required a customer character can be used as replacement
+			 */
+			continue;
+		} else {
+			rc = i2c_write_dt(&config->bus, text, len);
+			if (!rc) {
+				return rc;
+			}
+		}
```

Comment:
Not seeing any code here that updates the variables keeping a track of the current position